sdk/typescript/src/signers/signer.ts

```diff
@@ -11,8 +11,11 @@ import { PublicKey, SignatureScheme } from '../cryptography/publickey';
  */
 export type SignaturePubkeyPair = {
   signatureScheme: SignatureScheme;
-  signature: Uint8Array;
-  pubKey: PublicKey;
+  /** Base64-encoded signature */
+  signature: string;
+  /** Base64-encoded public key */
+  pubKey: string;
+  serializedSignature: string;
```

Comment:
is this just for convenience here? serializeSignature should already contain (signatureScheme, signature, pubKey)

Reply:
Yeah this is just convenience. This allows this signature + data to directly be submitted without another serialization step later, while simultaneously allowing the individual fields to be used as desired (and presented in a JSON-friendly format that we can send between different contexts like wallet / dapps).

Reply:
The alternative is just to return the `serializedSignature`, and then extract everything later on, it just means we need to convert data quite a bit which might be annoying. But I'm open to suggestions here if you think it'd be more worthwhile to just return that and then have utils to extract it out.

Reply:
I ended up just using the serialized signature and adding new `toSerializedSignature` and `fromSerializedSignature` utils to help with this!